src/Database/Query.php

```diff
@@ -28,7 +30,7 @@
  * for dynamically constructing each query part, execute it and transform it
  * to a specific SQL dialect.
  */
-class Query implements ExpressionInterface, IteratorAggregate
+class Query implements ExpressionInterface, IteratorAggregate, QueryInterface
```

Comment:
I'm not sure that the low level query object necessarily needs to implement the QueryInterface. It isn't really analogous to webservice queries, or elasticsearch - ORM\Query is though.